docs/Generators.md

```diff
@@ -369,3 +369,113 @@ spec:
 Any `config.json` files found under the `cluster-config` directory will be parameterized based on the `path` wildcard pattern specified. Within each file JSON fields are flattened into key/value pairs, with this ApplicationSet example using the `cluster.address` as `cluster.name` parameters in the template.
 
 As with other generators, clusters *must* already be defined within Argo CD, in order to generate Applications for them.
+
+
+
+
+## Matrix Generator
+The matrix generator combines two other generators by multiplying the parameters of them.
+
+### Use Case Example
+Imagine we have two clusters: 
+* staging (@ https://1.2.3.4)
+* production (@ https://2.4.6.8)
+
+And our application yamls are defined in a git repository:
+* argo workflows controller (examples/git-generator-directory/cluster-addons/argo-workflows)
+* prometheus operator (/examples/git-generator-directory/cluster-addons/prometheus-operator)
+
+Our goal is to deploy both application in both clusters. 
+For that we will use the matrix generator, with the git and the cluster as inner generators:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-git
+spec:
+  generators:
+    - matrix:
+        generators:
+            - git:
+                name: cluster-deployments
```

Comment:
No `name:` field on Git generator, think this can be removed? Plus need to add the `revision` field which is required.